sdk/include/opentelemetry/sdk/trace/sampler.h

```diff
@@ -49,7 +49,19 @@ class Sampler
 {
 public:
   // TODO: Remove this placeholder with real class
-  class SpanContext;
+  class SpanContext
```

Comment:
Could we move this into it's own file `api/include/opentelemetry/span_context.h`, with a remark that this is a placeholder for now? Then we could use `trace_api::SpanContext` below and have a stable API for samplers. Otherwise we will need to change the sampler interface and all `ShouldSample` signatures once we finally introduce `SpanContext`.

Reply:
We'll be addressing this in a following PR.